### ISSUE TITLE
resources and parameters must have unique names

### ISSUE THREAD

**Issue by PatMyron** [aws-cloudformation/cfn-python-lint#1689]
```yaml
Parameters:
  Name:
    Type: String
Resources:
  Name:
    Type: AWS::SNS::Topic
```
```bash
Template error: all resources and parameters must have unique names. Common name(s):[Name]
```

**Comment by kddejong** [comment]
Is that now an error in the service itself? I've been thinking about this for a while. It used to be that didn't error out (when I tested it).  I wanted to create it as a Warning since a Ref to the name could yield weird results.  I'm actually glad this is an error and this should be an easy rule to write.

**Comment by PatMyron** [comment]
Errors out in the service itself, think I'm almost done with the rule, just want to test before opening PR:
```python
"""
Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: MIT-0
"""
from cfnlint.rules import CloudFormationLintRule, RuleMatch


class Unique(CloudFormationLintRule):
    id = ''
    shortdesc = ''
    description = ''
    source_url = ''
    tags = ['parameters', 'resources']

    def match(self, cfn):
        matches = []
        for resource in cfn.get_resources():
            for parameter in cfn.template.get('Parameters', {}):
                if resource == parameter:
                    matches.append(RuleMatch(['Resources', resource], 'resources and parameters must not share name: ' + resource))
        return matches
```
